### PR TITLE
Fix project registry config conflict

### PR DESCRIPTION
**Pull request by Cappo**
This PR should fix the issue where setting a conflicting registry in the project `.npmrc` will no longer cause issues with authenticating against the publish registry. Audit command will still run against registry defined in project config as that will most likely point to where dependencies are pulled from. All other commands pertaining to publishing the package will have an explicit `--registry` flag set to make sure we are communicating with the publish registry when pertinent.

### CODE REVIEW

File: internal/npm/plugin.go [hunk 1]
```diff
@@ -71,6 +71,43 @@ func (p *plugin) Validate() error {
...
+
+	exists, osErr := p.os.Exists(".npmrc")
+	if osErr != nil {
+		return fmt.Errorf("failed when looking for local .npmrc: %w", osErr)
+	}
+
+	if exists {
+		log.Trace("Renaming local .npmrc")
+
+		osErr := p.os.Fs.Rename(".npmrc", ".tmp-npmrc")
```
Comment (GregoryDosh) [inline internal/npm/plugin.go:85]: Not sure which way makes more sense here but does it make sense to rename this file in the local workspace or to rename/move it to something like `/tmp/.npmrc/` and back. If things are manipulated within the workspace and the plugin fails or if this plugin is used in parallel (like in stages) could this risk some weird clobbering issue or something? Not sure if it matters but it's the only concern I had. Failed plugins would possibly leave the npmrc in a weird state.

**Review by GregoryDosh — APPROVED** [review]

File: internal/npm/plugin.go (hunk 1 above)
Comment (Cappo) [reply internal/npm/plugin.go:85]: Yeah, I had thought of that, but GoLang doesn't like moving files across devices out of the box. Got the following:
```
FATAL  [0000] failed to create rename local .npmrc: rename .npmrc /tmp/.npmrc: invalid cross-device link
```

Could look into seeing if there's a way to override that.

**Review by GregoryDosh — APPROVED** [review]
LGTM

**Review by plyr4 — APPROVED** [review]

**Review by wass3r — COMMENTED** [review]
minor nitpick since it's just for tests.. but, i prefer to use `example.com` that can't be owned by anyone for fake domains. i believe `test.com` is an actual domain that someone owns. unless this is intentional.

**Review by wass3r — APPROVED** [review]

**Pull request merged**